ethers-providers/src/transports/quorum.rs

```diff
@@ -160,23 +160,57 @@ impl<T> QuorumProviderBuilder<T> {
 }
 
 impl<T: JsonRpcClientWrapper> QuorumProvider<T> {
-    /// Returns the block height that _all_ providers have surpassed.
-    ///
-    /// This is the minimum of all provider's block numbers
-    async fn get_minimum_block_number(&self) -> Result<U64, ProviderError> {
-        let mut numbers = join_all(self.providers.iter().map(|provider| async move {
-            let block = provider.inner.request("eth_blockNumber", QuorumParams::Zst).await?;
-            serde_json::from_value::<U64>(block).map_err(ProviderError::from)
-        }))
-        .await
-        .into_iter()
-        .collect::<Result<Vec<_>, _>>()?;
-        numbers.sort();
-
-        numbers
-            .into_iter()
-            .next()
-            .ok_or_else(|| ProviderError::CustomError("No Providers".to_string()))
+    /// Returns the block height that a _quorum_ of providers have reached.
+    async fn get_quorum_block_number(&self) -> Result<U64, QuorumError> {
+        let mut queries = self
+            .providers
+            .iter()
+            .map(|provider| {
+                Box::pin(async move {
+                    let block =
+                        provider.inner.request("eth_blockNumber", QuorumParams::Zst).await?;
+                    serde_json::from_value::<U64>(block)
+                        .map(|b| (provider, b))
+                        .map_err(ProviderError::from)
+                })
+            })
+            .collect::<Vec<_>>();
+
+        let mut numbers = vec![];
+        let mut errors = vec![];
+        while !queries.is_empty() {
+            let (response, _index, remaining) = future::select_all(queries).await;
+            queries = remaining;
+            if response.is_ok() {
```

Comment:
purely for my own learning, is this pattern of using `is_ok` and unwrapping any less desirable than using a match or `if let`?

Reply:
No, I am just being an idiot.